### PR TITLE
fishnet: 2.2.4 -> 2.2.5

### PR DESCRIPTION
**Pull request by tu-maurice**
###### Motivation for this change

Based on #113828 but with the necessary upgrade of the assets.

Closes #113828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The assets don't build on darwin due to the usage of autoPatchelfHook. Can you mark it linux only?

**Comment by tu-maurice** [comment]
Sure. What would be the best way to handle the assets on darwin though? Could one just use the unpatched assets for darwin? E.g. not use autoPatchelfHook on darwin, but leave everything else the same? I sadly have no darwin machine to test this on.

**Comment by SuperSandro2000** [comment]
I think you need to do it manually but I am actually not sure but if you do not have a darwin machine to test this on i don't want to force you into maintaining it for it.

**Comment by tu-maurice** [comment]
Hm, yeah. If this would've been an easy fix I could as well just have added it, but this seems like more serious work to be done. Let's just leave it at linux only until someone with a darwin machine is willing to maintain this with me. Shall I set the main package linux only too or is it okay as long as one of the dependencies is linux only?

**Comment by SuperSandro2000** [comment]
> Shall I set the main package linux only too or is it okay as long as one of the dependencies is linux only?

Upstream mentions darwin support please mark it broken on darwin. If upstream never mentions darwin support please set platforms to Linux only.

**Comment by tu-maurice** [comment]
Hm. fishnet upstream progressed to version 2.2.6 in the meantime, however one of the dependencies for 2.2.6 now requires the newest rust stable 1.50.0 which is not merged yet (#112792). We should probably wait for the rust update and then update fishnet to 2.2.6 in one step.

File: pkgs/servers/fishnet/default.nix [hunk 1]
```diff
@@ -33,5 +33,6 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/niklasf/fishnet";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ tu-maurice ];
+    broken = !stdenv.isLinux || !(stdenv.isx86_64 || stdenv.isAarch64);
```
Comment (SuperSandro2000) [inline pkgs/servers/fishnet/default.nix:36]: Can we set platforms to x86_64-linux instead?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/fishnet/default.nix (hunk 1 above)
Comment (tu-maurice) [reply pkgs/servers/fishnet/default.nix:36]: Well, you told me earlier to mark it broken, but we sure can also just set the platforms instead.

**Pull request merged**